### PR TITLE
Issue 28

### PR DESCRIPTION
**Pull request by lutaylor**
**JIRA Ticket**: https://github.com/Islandora-Labs/islandora_vagrant_base_box/issues/28 (no JIRA just issue in queue)

# What does this Pull Request do?

Updates Fits to 1.1.0
Solr to 4.10.4
GSEARCH HEAD
gsearch_extensions 0.1.2

Some of the pathing has been updated to avoid having to change any files in basic-solr-config it should just work as is. No longer pulling pre-built GSEARCH configs from the York server since they wouldn't work after the updates. The ant process will now run when box is cut. 

# What's new?
Updated Fits, Solr, Gsearch, and Gsearch extensions. Also added log4j that will log gsearch errors to /usr/local/fedora/server/logs/fedoragsearch.log.

# How should this be tested?

Clone my repo. Bring up islandora_vagrant_base_box then halt and package. After packaging add the package.box

vagrant box add testbox package.box 

Then update config.vm.box = "testbox" in the islandora_vagrant Vagrantfile.

Then in islandora_vagrant

vagrant up

Try to ingest, modify and delete objects. Ensure that Solr is updating as expected. 

# Additional Notes:
Note: Drupal, PHP and other binaries will be updated when the new basebox is cut. Tomcat is currently installing from latest from Ubuntu repositories. Pulling it out to install from source is going to be a lot more work than first anticipated and should be handled under a seperate issue/PR if we feel that it is still necessary. Gsearch has been updated so it should be compatible with latest Tomcat when update takes place.

# Interested parties
@DonRichards
@Natkeeran

### CODE REVIEW

**Comment by lutaylor** [comment]
Updated to FITS 1.1.1 as per @ruebot  comment.

**Comment by DonRichards** [comment]
I'm little confused @lutaylor. Why vagrant box add testbox package.box ?

**Comment by lutaylor** [comment]
@DonRichards 
I added a seperate box so it wouldn't mess with the existing base box that was downloaded on my system. You don't necessarily have to do this but I figured it was the way to go to avoid possibly pulling the wrong box but I could be wrong.

**Comment by lutaylor** [comment]
@DonRichards 

We will also need this one done and a new basebox cut to be compatible.

**Comment by lutaylor** [comment]
@DonRichards 
Did you need anything from me to help get this merged/tested?

File: configs/variables [hunk 1]
```diff
@@ -13,9 +13,9 @@ FEDORA_HOME="/usr/local/fedora"
 FFMPEG_VERSION=1.1.4
 
 FITS_HOME="/usr/local/fits"
-FITS_VERSION=0.10.1
+FITS_VERSION=1.1.1
 
-SOLR_HOME="/usr/local/solr"
-SOLR_VERSION=4.2.0
+SOLR_HOME="/usr/local/fedora/solr"
```
Comment (DiegoPino) [inline configs/variables]: Hey People, why did you change the Solr home directory? Being it inside fedora is a bit confusing. Are there any strict devops requirements for this?

Comment (DiegoPino) [inline configs/variables]: @lutaylor I get it, those properties are set for the build in Solr. I would suggest you go for sed. It's good practice for people poking around in the machine to have the different services in their own directories, but mostly in the same place, they were available few months ago.

File: configs/log4j.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <level value="INFO" />
+    <appender-ref ref="FILEOUT"/>
+  </logger>
+  
+  <logger name="dk.defxws.fgslucene" additivity="false">
+    <level value="INFO" />
+    <appender-ref ref="FILEOUT"/>
+  </logger>
+
+  <logger name="dk.defxws.fgssolr" additivity="false">
```
Comment (DiegoPino) [inline configs/log4j.xml]: Is this gsearch 2.8? If so, we could add the extra solr remote log handler in case someone wants the internal one swapped and wonders why logs are not working ? 
```Shell
<logger name="dk.defxws.fgssolrremote" additivity="false">
<level value="LOGLEVEL" />
<appender-ref ref="FILEOUT"/>
</logger>````

Comment (lutaylor) [inline configs/log4j.xml]: This look good?
 https://github.com/lutaylor/islandora_vagrant_base_box/commit/65ae56ab067cb164aed53d748218674794d972c7

Gsearch_extensions were updated since I put the main PR in so I had to update the version number. 

Also see https://github.com/Islandora-Labs/islandora_vagrant/pull/128 updated to match.

**Comment by DiegoPino** [comment]
@lutaylor++ looks good, thanks for addressing my concerns.

**Comment by lutaylor** [comment]
@Islandora-Labs/committers 
Anyone willing to Merge?

**Comment by DiegoPino** [comment]
@lutaylor if you are ok with that, I can merge. Was giving DevOps IG folks preference.

**Comment by lutaylor** [comment]
@DiegoPino 
Sure that would be great. Unless someone else wants to chime in. 

We still need someone with access to the Hashicorp account to publish the new basebox once this and the other PR gets merged.  I do not have access.
@manez Do you have access? I think you and @ruebot were the only ones?

**Comment by lutaylor** [comment]
Thanks, @ruebot I will connect with @manez and let you know if I need help. 

@DiegoPino 

Sure go ahead and merge if you are able. Thank you for your help!

**Comment by lutaylor** [comment]
@ruebot  Sent you an email with a few questions whenever you get a moment.

**Comment by lutaylor** [comment]
The new basebox has been published. Confirmed that I can delete/add solr objects fine. Please let me know if you find any issues.
